src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -369,21 +429,36 @@ private void conjureSnapshotTablesFromRowIds(SnapshotRequestRowIdModel requestRo
                     mapColumnList.add(snapshotMapColumn);
                 });
         }
+        // pass through all of the relationships that touch request tables from the dataset into the snapshot
+        List<Relationship> sourceRelationships = dataset.getRelationships()
```

Comment:
Why not use this algorithm for all snapshots? It has the same name issue as the other algorithm.

Reply:
I think it would be confusing to include relationships for tables/columns that don't exist in the snapshot, so that's why I have three separate ways of picking which relationships go in. Here it makes sense for all to go in. In the rowId case, only relationships that match mentioned tables+columns go in. In the asset case only relationships mentioned in the asset go in -- I would be surprised if a "rogue" relationship that I didn't mention in my asset showed up in the resulting snapshot.

I do think the naming thing can be solved separately than these, will take a look.

Reply:
My concern is that the relationships needed for building the snapshot are a subset of the relationships that might be interesting to use the snapshot. imagine going from participant to sample to aliquot. There might be a relationship between participant and aliquot, but it would not be present in the asset because we would not want to have two paths to walk from participant. So i don't think that just using the asset relationships is the best we can do.

Reply:
We done may need more than one algorithm for different types of snapshots. I may not understand what code is actually doing here.

Reply:
Ah good point! Makes sense to not confuse the `follows` list with the relationships that should go into the snapshot. Okay I've generalized the algorithm so that it picks all of the relationships that "cover" the snapshot (where every relationship has a from/to table + column in the snapshot).